File: backend/blackbox/src/main/java/com/insight_web/blackbox/adapters/postgres/site/SiteEntity.java
package com.insight_web.blackbox.adapters.postgres.site;

import com.insight_web.blackbox.domain.Site;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "site")
public class SiteEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private String id;
    private Long ownerId;
    private String siteName;

    public Site toDomainModel() {
        var site = new Site();
        site.setId(id);
        site.setSiteName(siteName);
        site.setOwnerId(ownerId);
        return site;
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/adapters/postgres/site/SiteEntity.kt
package com.insightweb.auth.adapters.postgres.site

import com.insightweb.auth.domain.Site
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "site")
class SiteEntity {
    @Id
    var id: String? = null
    var ownerId: Long? = null
    var siteName: String? = null

    fun fillForSaving(site: Site): SiteEntity = apply {
        id = site.id
        ownerId = site.ownerId
        siteName = site.siteName
    }

    fun toDomainModel(): Site = Site(
        id = requireNotNull(id),
        ownerId = requireNotNull(ownerId),
        siteName = requireNotNull(siteName)
    )
}